src/garage/torch/policies/gaussian_mlp_policy.py

```diff
@@ -0,0 +1,139 @@
+import torch
+from torch import nn
+
+from garage.torch.modules import GaussianMLPModule
+from garage.torch.policies import Policy
+
+class GaussianMLPPolicy(nn.Module, Policy):
```

Comment:
please provide a more modular way of defining (several types) of GaussianMLPModels.

if you look closely, this module implements 3-5 different models which switch between configurations based on the constructor. instead, we can just have 3-5 different classes which mostly rely on the same base class.

@lywong92 has a similar idea going on in #764. I want to avoid creating technical debt in torch which we are trying to fix in https://github.com/rlworkgroup/garage/issues/587

your implementation can serve as a template for the TF version.

Reply:
So... you means dividing the current GaussianMLPModule into 3 different classes?
for example, GaussianSharedMlpModule, GaussianAdaptiveMlpModule, GaussianMlpModule?

Reply:
correct. but you should also provide concrete implementation which takes parameters, and produces a policy by constructing the correct modules and passing them.

Reply:
I changed the GaussianMLPModules (GaussianMLPModule, GaussianMLPAdaptiveModule, GaussianMLPSharedModule) and GaussianMLPPolicy.

Could you check the changes?

Reply:
checked